utils/src/main/scala/za/co/absa/enceladus/utils/unicode/ParameterConversion.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.unicode
+
+object ParameterConversion {
+
+  implicit class UnicodeValueString(s: String) {
+    val unicodeElement = """[uU]?\+?([0-9a-fA-F]{4})$""".r
+
+    def includingUnicode: String = s match {
+      case unicodeElement(hex) => Integer.parseInt(hex, 16).toChar.toString
+      case other => other
+    }
+  }
+
+  implicit class NoneValueString(s: String) {
+    def includingNone: String = s.toLowerCase() match {
+      case "none" => ""
+      case _ => s
```

Comment:
I think the case should also check that the string is exactly 1 character long (or at most 1 char long - do we allow empty string or the only possibility to express empty char is by `none`?) and fail otherwise. And the class/method name could also reflect this, IMHO.

... but thinking about it, using
```someString.includingUnicode``` would then go unchecked, if the length check was only in `includedNone`

It seems that these two are always used together. If that is always going to be the case, maybe a better suggestion is to allow API like:

```scala
someString.toUnicodeOrChar(includingNone = true)
```
That could work for:
`a`, `u1234`, `U+1234`, `none` and fail for `abc`, `U123`, ...

Just a suggestion, though.

Reply:
The idea is that it is still currently possible to specify empty string. The none keyword is used in case the usage is from a scripting environment where ''(double single quote) would be ignored. 
Furthermore, Spark checks if there is at most one character, so values like `abc` would fail from Spark side. 

About using them together, I am open to suggestions. My idea was to keep them separate in case  some further usage would use just the None part without coupling it to the unicode part.

Reply:
What you say makes sense. On the other hand, to make the `includingNone` truly versatile, its signature should in my opinion probably be
```
implicit class NoneValueString(s: String) {
    def includingNone: Option[String] = s.toLowerCase match {
      case "none" => None
      case _ => Some(s)
}
```
Nevertheless, all these are just suggestions. The current code is acceptable and it looks like works as expected. 👍

Reply:
The thing is that if None is passed as a csv option, then it means that the default values will be used, which will be `,`(comma) for delimiter,  `\`(backslash) for escape and `"` for quote. The value that we want to pass in the None case is `""`(empty string) or `u0000`